### PR TITLE
Notification Config Backend

### PR DESCRIPTION
**Pull request by hedgehog1833**
The Notification Config now has its own repository, service and controller. On the controller level I use 2 controller (one for the config, one for the notification itself) to have this separated but on the Service level i kept one service for simplicity.

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Make Notifications configurable](https://trello.com/c/hlDkMGvu/255-make-notifications-configurable)

**Comment by hedgehog1833** [comment]
I just added the deletion of the notification config when the user is deleted.

File: persistence/src/main/java/rocks/metaldetector/persistence/domain/notification/NotificationConfigEntity.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  @OneToOne(targetEntity = UserEntity.class)
+  @JoinColumn(nullable = false, name = "users_id")
+  private UserEntity user;
+
+  @Column(name = "notify", nullable = false, columnDefinition = "boolean default false")
+  @Setter
+  @Builder.Default
+  private Boolean notify = false;
+
+  @Column(name = "frequencyInWeeks", nullable = false, columnDefinition = "integer default 4")
```
Comment (DanielW1987) [inline persistence/src/main/java/rocks/metaldetector/persistence/domain/notification/NotificationConfigEntity.java:35]: for column names we use underscores insted of camel case

**Comment by hedgehog1833** [comment]
On that occasion: we have other stuff that is probably currently not deleted, when the user is deleted (spotify authorization, followActions). I'll create a card for that.

File: webapp/src/main/java/rocks/metaldetector/service/notification/NotificationConfigDtoTransformer.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  public NotificationConfigEntity transform(NotificationConfigDto notificationConfigDto) {
+    return NotificationConfigEntity.builder()
+        .notify(notificationConfigDto.isNotify())
+        .notificationAtReleaseDate(notificationConfigDto.isNotificationAtReleaseDate())
+        .notificationAtAnnouncementDate(notificationConfigDto.isNotificationAtAnnouncementDate())
+        .frequencyInWeeks(notificationConfigDto.getFrequencyInWeeks())
+        .build();
+  }
+}
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/service/notification/NotificationConfigDtoTransformer.java:17]: Do we need this?

Comment (hedgehog1833) [reply webapp/src/main/java/rocks/metaldetector/service/notification/NotificationConfigDtoTransformer.java:17]: Ahh no we don't. That's another hickup from my local merge chaos 😅

**Review by DanielW1987 — APPROVED** [review]

**Pull request merged**